src/views/partials/breaches.js

```diff
@@ -32,10 +32,17 @@ function createBreachRows (data) {
         dataClasses: longList.format(dataClassesTranslated)
       })
 
+      const logo = logos.has(breach.Domain)
+        ? `<img src='${logos.get(breach.Domain)}' alt='' class='breach-logo' height='32' />`
+        : `<span role="img" aria-hidden='true' class='breach-logo' style='background-color: var(${getColorForName(breach.Domain)});'>${breach.Domain.substring(0, 1)}</span>`
```

Comment:
Not sure if I'm tracking all the code properly here, but does `breach.Domain` here account for potentially empty breach domains? Currently 26 breaches have an empty domain, per:

```sh
curl https://monitor.firefox.com/hibp/breaches | jq '.[] | [{name:.Name, domain:.Domain}]' | grep -B1 '"domain": ""'
```

it almost looks like we check if the logos Map has the current `breach.Domain`. Presumably if the logo is NOT in that map, there was an empty `Domain` or DDG just didn't have a hit. If there WAS a hit, we return the found logo. If not, we might be trying to calcuate a color via `getColorForName("")` on an empty string and trying to do `breach.Domain.substring(0, 1)` on an empty string (which would return an empty string). Not sure if we need to maybe fall back to `breach.Name` or something if Domain is an empty string.

Reply:
Oh, thanks for sharing the link to the data structure! And yes, good point, apparently the domains are empty strings if they're not set... Looking through the `breach.Name`s in the data set, those do appear to be properly set for all breaches, so I just used that to determine the colour now - the actual colour doesn't matter that much, as long as it's deterministic for every breach. de467e27